build.gradle

```diff
@@ -123,7 +123,7 @@ assets {
 
 bootRun {
     addResources = true
-    dependsOn 'npm_run_bundle'
+    //dependsOn 'npm_run_bundle'
```

Comment:
Ignore for now, I just didn't want to wait around for this shit while I was running tests.

Reply:
Apologies in advance but we don't have a place to put these tangential discussions so I'm adding my thoughts here until I can find a good place for this. 

---

@awalkowiak (or anyone else who might know) Is it possible to remove "npm run bundle" when using run-app or debug-app. I think we need to execute it for packaging the WAR file, but when running the application in developer we can start the React / webpack command ourselves.

In fact, if I was a developer I'd probably prefer to use 
```
npm run watch
```
on its own. 

However, I know without the npm_run_bundle the React application won't be available or not having it bundled each time will cause confusion if we're using older resources. But the start is so painful I'd be ok with it. 

A few followup questions and comments
* Is there a good reason to keep the npm_run_bundle dependency on bootRun i.e. grails run-app? 
* Is there a way for us to run the "dependsOn" process in the background? 
* Can we add a placeholder bundle that writes to an error message to the DOM or an alert stating "The app is unavailable - please execute `npm run watch` in order to access the UI"
* We could probably add a command line argument (property) to enable/disable webpack mode (i.e. bundle, watch, none) with none being the default 
```
gradlew bootRun -Pmode=watch
```
We'd also want to figure out the equivalent when running via Grails since I don't think Grails passes arguments to gradle by default. We would need to explicitly capture and pass the arguments to Gradle.
```
grails run-app
```

Here's a resource that I think Artur probably encountered when working on the React integration a long time ago. 
https://stackoverflow.com/questions/40348588/is-it-possible-to-integrate-react-using-webpack-with-grails

And at the bottom, there's a suggestion I don't love, but also don't hate re: npm scripts. It doesn't really help us unless we decided to fundamentally change the way we start the application in development. Although I guess having both options (heavy version using grails/gradle and light version using npm scripts) might be fine.
https://stackoverflow.com/a/40349256/136597

Reply:
@jmiranda I had no chance to dive into this one more, perhaps this is a candidate for a discovery ticket?

Reply:
I'm about to revert this change but wanted to get some data before doing so. 

Can you all provide some metrics from your development environments (CPU usage, build time) when running "npm run bulde"

Without "npm run bundle" being executed, the build takes about 90 seconds and I can still use my computer while `grails run-app` command is starting up the application

With "npm run bundle", load is high and CPU is pegged at 100% for about 5 minutes making my laptop unuseable during that time. In fact, I can't even finish writing this message.  

```
top - 12:00:17 up 2 days,  3:11,  1 user,  load average: 9.34, 11.36, 8.01
Tasks: 476 total,   1 running, 475 sleeping,   0 stopped,   0 zombie
%Cpu(s): 85.0 us,  6.1 sy,  0.0 ni,  7.6 id,  0.0 wa,  0.0 hi,  1.3 si,  0.0 st
MiB Mem :  31730.5 total,   3986.2 free,  20225.6 used,   7518.7 buff/cache
MiB Swap:  15737.5 total,  15630.6 free,    106.9 used.  10243.8 avail Mem 

    PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND                                                                                                                               
 269879 jmiranda  20   0   33.3g   1.1g  34336 S 569.2   3.7   4:53.21 webpack                                                                                                                               
 269776 jmiranda  20   0 5192956 875540  35612 S  75.2   2.7   1:57.32 java                                                                                                                                  
  17317 jmiranda  20   0 1133.9g 448144 118096 S  27.5   1.4 358:23.45 Logseq                                                                                                                                
   4396 jmiranda  20   0 5383544 817812 106552 S  12.9   2.5 116:05.66 gnome-shell                                                                                                                           
  13755 jmiranda  20   0   33.1g 724856 300296 S  11.6   2.2 235:45.79 chrome                                                                                                                                
  14246 jmiranda  20   0 1138.3g 273436 137364 S  10.6   0.8  43:57.59 chrome                         ```
```

I would like to know whether this is anomaly or whether others are experiencing this. If it's just me I'll keep commenting out npm_run_bundle on my own. If others are encountering this behavior, I'll create a discovery ticket to see if there's something we can do about it.


cc @EWaterman @kchelstowski @alannadolny @drodzewicz

Reply:
I get a pretty decent spike up as well, though it doesn't totally brick my computer

![image](https://github.com/openboxes/openboxes/assets/6844376/e57a32ca-dd00-4a13-9251-2ca114d23233)